client/signup/main.jsx

```diff
@@ -460,7 +460,9 @@ class Signup extends React.Component {
 				( this.props.signupDependencies &&
 					this.props.signupDependencies.domainItem &&
 					this.props.signupDependencies.domainItem.is_domain_registration &&
```

Comment:
I believe we set the `domainItem` in `signupDependencies` too. Can probably do:

( this.props.signupDependencies.domainItem.is_domain_registration || this.props.signupDependencies.domainItem.product_slug === 'domain_transfer' )

Reply:
Even better, can probably use https://github.com/Automattic/wp-calypso/blob/fd8135f704a8681269ee48876c185e2fbbb85ad8/client/lib/products-values/index.js#L251. Not 100% sure tho, but could work. Otherwise, use the constants for `domain_transfer` slug https://github.com/Automattic/wp-calypso/blob/1b7cff8b4a09072393fdbe3733bd7f4ac1edcb1d/client/lib/domains/constants.js#L74.